airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/check/python/commands.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import logging
+import sys
+from typing import Optional
+
+import asyncclick as click
+import dagger
+from pipelines.airbyte_ci.format.consts import DEFAULT_FORMAT_IGNORE_LIST
+from pipelines.cli.click_decorators import click_ignore_unused_kwargs
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.contexts.click_pipeline_context import ClickPipelineContext, pass_pipeline_context
+
+
+@click.command()
+@pass_pipeline_context
+@click_ignore_unused_kwargs
+async def python(ctx: ClickPipelineContext):
+    """Format python code via black and isort."""
+    success = await format_python(ctx)
+    if not success:
+        sys.exit(1)
+
+
+async def format_python(ctx: ClickPipelineContext) -> bool:
+    """Checks whether the repository is formatted correctly.
+    Args:
+        fix (bool): Whether to automatically fix any formatting issues detected.
+    Returns:
+        bool: True if the check/format succeeded, false otherwise
+    """
+    logger = logging.getLogger(f"format")
+
+    dagger_client = ctx.params["dagger_client"]
+    try:
+        format_container = await (
+            dagger_client.container()
+            .from_("python:3.10.13-slim")
+            .with_env_variable("PIPX_BIN_DIR", "/usr/local/bin")
+            .with_exec(
+                sh_dash_c(
+                    [
+                        "apt-get update",
+                        "apt-get install -y bash git curl",
+                        "pip install pipx",
+                        "pipx ensurepath",
+                        "pipx install poetry",
+                    ]
+                )
+            )
+            .with_mounted_directory(
+                "/src",
+                dagger_client.host().directory(
+                    ".", include=["**/*.py", "pyproject.toml", "poetry.lock"], exclude=DEFAULT_FORMAT_IGNORE_LIST
+                ),
+            )
+            .with_workdir(f"/src")
+            .with_exec(["poetry", "install"])
```

Comment:
This is going to slow things down because we're going to run this every time. Do we really need poetry here to run isort and black? I'd rather we pip installed some pinned hardcoded version of isort and black _before_ mounting the source files.

Reply:
I see what you mean. The idea here is to keep the configuration for black/isort in the base repository-level `pyproject.toml` so that it is easy to configure your IDE to autoformat for you. This was @bnchrch's idea, and I think its a good one! 

That being said, if there's a different way to read the versions from that and install it directly (would that be an antipattern?) Id be in for it. In the worst case at least we can create a group for them so that we only install these 2 dependencies.